File: 2260-divide-a-string-into-groups-of-size-k/2260-divide-a-string-into-groups-of-size-k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        i=0
        ans = []
        while i+k < len(s):
            ch = s[i:i+k]
            ans.append(ch)
            i = i+k
        ch = ""
        while i < len(s):
            ch += s[i]
            i+=1
        while(len(ch) < k):
            ch += fill
        ans.append(ch)
        return ans
        

File: 2260-divide-a-string-into-groups-of-size-k/2260-divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string> ans;
        int i = 0;
        while(i < s.size()){
            string ch = "";
            for(int j=i;j<i+k and j<s.size();j++){
                ch += s[j];
            }
            if(ch.size()<k){
                ch += string(k - ch.size(), fill);
            }
            ans.push_back(ch);
            i+=k;

        }
        return ans;
        
    }
};